zerver/openapi/openapi.py

```diff
@@ -282,3 +297,24 @@ def likely_deprecated_parameter(parameter_description: str) -> bool:
         return True
 
     return "**Deprecated**" in parameter_description
+
+SKIP_JSON = {'/fetch_api_key:post'}
+
+def validate_request(url: str, method: str, data: Dict[str, Any],
+                     http_headers: Dict[str, Any], json_url: bool,
+                     uses_invalid_parameters: bool) -> None:
+    # Some JSON endpoints have different parameters compared to
+    # their `/api/v1` counterparts.
+    if json_url and url + ':' + method in SKIP_JSON:
+        return
+    # Currently it does not support endpoints with file uploads
+    if url == '/user_uploads' or url.startswith('/realm/emoji/'):
+        return
+    mock_request = MockRequest('http://localhost:9991/', method, '/api/v1' + url,
+                               headers=http_headers, args=data)
+    result = openapi_spec.core_validator().validate(mock_request)
+    # If it is valid then error list should be empty
+    if uses_invalid_parameters:
+        assert(len(result.errors) != 0)
+    else:
+        result.raise_for_errors()
```

Comment:
What does the error output for this look like?  We absolutely have to make sure the error message mentions the `uses_invalid_parameters` option.

Reply:
Changed the error message accordingly.